### PR TITLE
Added configuration option for not doing chunked upload for HTTP remotes

### PR DESCRIPTION
**Pull request by scotty3005**
Changes to be committed:
	modified:   dvc/config_schema.py
	modified:   dvc/fs/http.py

Draft proposal for feature request https://github.com/iterative/dvc/issues/6118

* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/fs/http.py [hunk 1]
```diff
@@ -159,6 +160,17 @@ def chunks(fobj):
         if response.status_code not in (200, 201):
             raise HTTPError(response.status_code, response.reason)
 
+    def _upload_fobj_not_chunked(self, fobj, to_info):
+        response = self.request(self.method, to_info.url, files={"file": fobj})
+        if response.status_code not in (200, 201):
```
Comment (karajan1001) [inline dvc/fs/http.py:165]: Duplicated code with what in `_upload_fobj_chunked` maybe we can move them to `_upload_fobj`

File: dvc/fs/http.py [hunk 2]
```diff
@@ -128,7 +129,7 @@ def _head(self, url):
         # Sometimes servers are configured to forbid HEAD requests
         # Context: https://github.com/iterative/dvc/issues/4131
         with self.request("GET", url, stream=True) as r:
-            if r.ok:
+            if r.ok or r.status_code == 404:
```
Comment (karajan1001) [inline dvc/fs/http.py:132]: Would this cause some problems when the server was actually not found (404). Some additional check are needed.

**Review by karajan1001 — CHANGES_REQUESTED** [review]

**Comment by karajan1001** [comment]
Thanks for your contributions, so this PR is related to #6118 ? I recommend You can add a link to it.

**Review by karajan1001 — COMMENTED** [review]
And, some tests are needed for this PR.

**Comment by scotty3005** [comment]
Hi, thanks for the feedback. I am willing to work on implementing some tests where needed. However I would be glad if I could have some help as it is not clear to me how to test remotes.

**Comment by efiop** [comment]
I also wonder if https://github.com/intake/filesystem_spec/blob/master/fsspec/implementations/http.py is already able to do what you need. We are migrating to fsspec and will be considering replacing our http implementation with their soon, so might be a good time to give it a shot.

**Comment by isidentical** [comment]
It seems like `open()` on `HTTPFileSystem` is read-only;
https://github.com/intake/filesystem_spec/blob/61ec18fb0a77d3a524a6b2d72340af3748c5bf2a/fsspec/implementations/http.py#L279-L280

**Comment by efiop** [comment]
@scotty3005 Note that your PR now includes perm changes to a lot of files, which obviously shouldn't be there. There is something wrong with your environment, most likely.

**Comment by scotty3005** [comment]
Hey @efiop, yes I noted the issue with the changes. I do not what happen. W.r.t. the authorship I believe it is because I worked from two different machines, and forgot to set the local user name and email. 
 In the meanwhile that you guys do the migration, would you prefer to close this and me to resubmit a cleaner PR (correcting the changes and authorship)?

**Comment by efiop** [comment]
@scotty3005 As @karajan1001 noted, the current logic will cause problems for other servers, so we need more research here to see if we can actually detect this and handle the fallback properly. Unfortunately, I would probably recommend closing this PR for now and going through the research to implement proper logic. 🙁

**Comment by scotty3005** [comment]
Hey @efiop, sure I understand. I will close this and keep maintaining the fork for my particular use case. Thanks for the time you guys dedicated to this issue. I am looking forward to contribute to DVC in other occasions.

**Pull request closed without merging**